compiler/rustc_codegen_llvm/src/gotoc/metadata.rs

```diff
@@ -394,6 +394,37 @@ impl<'tcx> GotocCtx<'tcx> {
 
         Expr::statement_expression(body, t).with_location(loc)
     }
+
+    /// RMC-specific function to sanity check expected components of in code
+    /// generation. If users see these assertions fail, something in our
+    /// translation to Gotoc has gone wrong, and we want them to file an issue.
+    pub fn codegen_sanity_check(
```

Comment:
I don't think this uses `self` at all.  Make it a compound constructor on `goto_program::Stmt`